h/api/views.py

```diff
@@ -76,9 +76,18 @@ def index(context, request):
 @api_config(context=Root, name='search')
 def search(request):
     """Search the database for annotations matching with the given query."""
+    search_normalised_uris = request.feature('search_normalised')
```

Comment:
Yeah, so this is ugly, and it's what I meant when I said the other day that some feature flags are really technical debt, in the sense that they allow us to go faster now (by deploying things before we're certain we're ready to release them) at the cost of having to do some work later to clean them up (i.e. remove the feature flag and the other branch of the code).

In this case, I think the benefits (we can test the new URI normalisation code before releasing it to the general public) outweigh the costs. And yes, it can be usefully feature flagged -- because:

1. `source_normalised` fields are added regardless of the flag
2. we can run the data migration without harming the current setup
3. requests to the API are authenticated, so we can switch on the new search code for staff only in testing